apps/communications/contacts/js/activities_vcard.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+/* global ActivityHandler, LazyLoader, VCardReader, Contacts */
+
+ActivityHandler.vCardActivityHandler = activity => {
+  var deps = '/shared/js/contacts/import/utilities/vcard_reader.js';
+  LazyLoader.load(deps, function() {
+    getvCardReader(activity.source.data.blob).then(readvCard);
+  });
+
+  function getvCardReader(blob) {
+    return new Promise((resolve) => {
+      var fileReader = new FileReader();
+      fileReader.readAsText(blob);
+      fileReader.onloadend = function() {
+        resolve(new VCardReader(fileReader.result));
+      };
+    });
+  }
+
+  function readvCard(vCardReader) {
+    var firstContact;
+    var cursor = vCardReader.getAll();
+    cursor.onsuccess = function(event) {
+      var contact = event.target.result;
+      // We check if there is only one contact to know what function
+      // we should call. If not, we render the contacts one by one.
+      if (contact) {
+        if (!firstContact) {
+          firstContact = contact;
+          cursor.continue();
+        } else {
+          renderContacts([firstContact, contact], activity, cursor);
+        }
+      } else if (firstContact) {
+        renderOneContact(firstContact, activity);
```

Comment:
instead of calling the ActivityHandler from this module, resulting in an inversion of authority, I would suggest that  we pass the handler as a parameter as I have suggested above